extras/ping/ping.go

```diff
@@ -307,10 +321,15 @@ func (p *Pinger) Run(ctx context.Context) (err error) {
 }
 
 func (p *Pinger) run(conn net.Conn) error {
-
+	if !p.sharedConnection {
+		defer p.conn.Close()
+	}
 	defer p.finish()
+	if p.Count == -1 || p.Count == 0 {
```

Comment:
```suggestion
	if p.Count <= 0 {
```

This is more robust because it excludes from the range all the invalid values, rather than just some of them